src/commands/login.rs

```diff
@@ -179,14 +187,18 @@ impl LoginCommand {
             self.status,
             self.get_device_code,
             &self.check_device_code,
+            &self.hippo_token,
         ) {
-            (true, false, false, None) => self.run_list().await,
-            (false, true, false, None) => self.run_status().await,
-            (false, false, true, None) => self.run_get_device_code().await,
-            (false, false, false, Some(device_code)) => {
+            (true, false, false, None, None) => self.run_list().await,
+            (false, true, false, None, None) => self.run_status().await,
+            (false, false, true, None, None) => self.run_get_device_code().await,
+            (false, false, false, Some(device_code), None) => {
                 self.run_check_device_code(device_code).await
             }
-            (false, false, false, None) => self.run_interactive_login().await,
+            (false, false, false, None, Some(fermyon_token)) => {
+                self.login_using_fermyon_token(fermyon_token).await
+            }
+            (false, false, false, None, None) => self.run_interactive_login().await,
```

Comment:
This is in dire need of refactoring, but doesn't need to happen here.

Reply:
we moved login under `run_interactive_login`, so this change is stale now